src/benchmarklib/benchmark_runner.cpp

```diff
@@ -0,0 +1,449 @@
+    const auto time_per_query = duration_ns / query_result.num_iterations;
+
+    nlohmann::json benchmark{
+        {"name", name},
+        {"iterations", query_result.num_iterations},
+        {"real_time", time_per_query},
+        {"cpu_time", time_per_query},
+        {"items_per_second", items_per_second},
+        {"time_unit", "ns"},
+    };
+
+    benchmarks.push_back(benchmark);
+  }
+
+  nlohmann::json report{{"context", _context}, {"benchmarks", benchmarks}};
+
+  stream << std::setw(2) << report << std::endl;
+}
+
+BenchmarkRunner BenchmarkRunner::create_tpch(BenchmarkConfig config, const std::vector<QueryID>& query_ids,
+                                             const float scale_factor) {
+  NamedQueries queries;
+  queries.reserve(query_ids.size());
+
+  for (const auto query_id : query_ids) {
+    queries.emplace_back("TPC-H " + std::to_string(query_id), tpch_queries.at(query_id));
+  }
+
+  config.out << "- Generating TPCH Tables with scale_factor=" << scale_factor << "..." << std::endl;
+
+  ColumnEncodingSpec encoding_spec{config.encoding_type};
+  const auto tables = opossum::TpchDbGenerator(scale_factor, config.chunk_size).generate();
+
+  for (auto& table : tables) {
+    if (config.encoding_type != EncodingType::Unencoded) {
+      ChunkEncoder::encode_all_chunks(table.second, encoding_spec);
+    }
+
+    StorageManager::get().add_table(tpch_table_names.at(table.first), table.second);
+  }
+  config.out << "- Done." << std::endl;
+
+  auto context = _create_context(config);
+  // Add TPCH-specific information
+  context.emplace("scale_factor", scale_factor);
+
+  return BenchmarkRunner(std::move(config), queries, context);
+}
+
+BenchmarkRunner BenchmarkRunner::create(BenchmarkConfig config, const std::string& table_path,
+                                        const std::string& query_path) {
+  const auto tables = _parse_table_path(table_path);
+  if (tables.empty()) {
+    throw std::runtime_error("No tables found in '" + table_path + "'");
+  }
+
+  CsvMeta csv_meta{};
+  csv_meta.chunk_size = config.chunk_size;
+
+  ColumnEncodingSpec encoding_spec{config.encoding_type};
+
+  for (const auto& table_str : tables) {
+    const auto table_name = fs::path{table_str}.stem().string();
+
+    std::shared_ptr<Table> table;
+    if (boost::algorithm::ends_with(table_str, ".tbl")) {
+      table = load_table(table_str, config.chunk_size);
+    } else {
+      table = CsvParser{}.parse(table_str, csv_meta);
+    }
+
+    if (config.encoding_type != EncodingType::Unencoded) {
+      ChunkEncoder::encode_all_chunks(table, encoding_spec);
+    }
+
+    StorageManager::get().add_table(table_name, table);
+    config.out << "- Adding table '" << table_name << "'" << std::endl;
+  }
+
+  const auto queries = _parse_query_path(query_path);
+  return BenchmarkRunner(config, queries, _create_context(config));
+}
+
+std::vector<std::string> BenchmarkRunner::_parse_table_path(const std::string& table_path) {
```

Comment:
I wouldn't call this "parsing". How about `read_table_folder`? Or `process_...`?